geps/gep-3155/index.md

```diff
@@ -0,0 +1,205 @@
+  // backends with TLS.
+  BackendTLS GatewayBackendTLS `json:"backendTLS,omitempty"'
+}
+
+type GatewayBackendTLS struct {
+  // ClientCertificateRef is a reference to an object that contains a Client
+  // Certificate.
+  //
+  // References to a resource in different namespace are invalid UNLESS there
+  // is a ReferenceGrant in the target namespace that allows the certificate
+  // to be attached. If a ReferenceGrant does not allow this reference, the
+  // "ResolvedRefs" condition MUST be set to False for this listener with the
+  // "RefNotPermitted" reason.
+  //
+  // ClientCertificateRef can reference to standard Kubernetes resources, i.e.
+  // Secret, or implementation-specific custom resources.
+  //
+  // This setting can be overriden on the service level by use of BackendTLSPolicy.
+  ClientCertificateRef SecretObjectReference `json:"clientCertificateRef,omitempty"`
+}
+```
+
+**2. Add a new `Client` field at the top level of BackendTLSPolicy**
+
+```go
+type BackendTLSPolicySpec {
+  // Client specifies overriden TLS settings for the given service.
+  //
+  // If Client is omitted, backend mTLS settings inherited from the given gateway are
+  // used.
+  Client TLSClientSettings `json:"client"`
+}
+
+type TLSClientSettings struct {
+  // CertificateRef is a reference to an object that contains a Client
+  // Certificate.
+  //
+  // References to a resource in different namespace are invalid UNLESS there
+  // is a ReferenceGrant in the target namespace that allows the certificate
+  // to be attached. If a ReferenceGrant does not allow this reference, the
+  // "ResolvedRefs" condition MUST be set to False for this listener with the
+  // "RefNotPermitted" reason.
+  //
+  // CertificateRef can reference to standard Kubernetes resources, i.e.
+  // Secret, or implementation-specific custom resources.
+  //
+  // If CertificateRef is not specified, no client certificate is presented when
+  // connecting to the service.
+  CertificateRef SecretObjectReference `json:"certificateRef,omitempty"`
+}
+```
+
+#### Limitations
+
+Configuring client certificate on the service level may result in multiple Gateways
+from different vendors sharing the same identity when connecting to a single service.
+
+The problem could be partially mitigated by allowing shared policies to selectively 
+target specific Gateway instances, GatewayClasses or gateway namespaces.
+
+### SANs on BackendTLSPolicy
+
+This change enables the certificate to have a different identity than the SNI
+(both are currently tied to the hostname field). This is particularly useful
+when using SPIFFE, which relies on URI Subject Names which are not valid SNIs 
+as per https://www.rfc-editor.org/rfc/rfc6066.html#section-3.
+
+In such case either connection properties or an arbitrary SNI, like cluster-local 
+service name could be used for certificate selection, while the identity validation
+will be done based on SubjectAltNames field.
```

Comment:
Again, I think we may need some examples or use cases here to illustrate how this would work, as I _think_ I get it, but am not sure.

Reply:
Maybe I should be clearer here. Currently, what this field is actually used for is not defined at all. Being able to specify more SANs is great, but as an implementer, what do I do with this field?

From the below struct docs, it seems like the answer is maybe "The serving certificate received when the Gateway implementation makes a client TLS connection to the backend must have at least one of the specified SANs set".

This is _critical_, because if we go ahead with also adding client certificates to this object, we now have both client and serving certificates on the same object. So it's really, really important that we leave no wiggle room here in the godoc.

Reply:
+1, good points @youngnick. I think the godoc below describes the behavior fairly clearly, maybe it would be helpful to pull that up to this paragraph as well?

> When specified, at least one of certificate's Subject Alternate Names MUST match at least one of the specified SubjectAltNames.

Reply:
Added this sentence, does it look better now?

Reply:
Sorry, but I think we really need the mention that it's the _serving_ cert that must be matched against.

So I think that sentence should be:

`When specified, the certificate served from the backend MUST have at least one Subject Alternate Name matching one of the specified SubjectAltNames`.